Disabled Icon Viewer Snippet

killingerm opened the pull request:
This draft includes `Snippet384` to inspect the behavior of the current icon-disabling algorithms implemented in the constructor of `org.eclipse.swt.graphics.Image(Device, Image, int)` when called with the `SWT.IMAGE_DISABLE` flag.

Currently, there are **two** different algorithms depending on whether Windows/MacOS or Linux is used. The Windows/MacOS implementation is outdated, and the generated disabled icons often look unappealing.

The snippet includes two new algorithms that can serve as a foundation for deriving a new algorithm, with adjustable parameters. This unified approach could also be applied to the Eclipse icons provided in disabled state, ensuring consistent appearance across all disabled icons, whether included in Eclipse JARs or generated dynamically using the `Image` constructor.

### How to Use
1. Select a folder containing images.
2. Adjust the sliders to modify the parameters of the new algorithms.

### Example Images
I have included example images that demonstrate the weaknesses of the current algorithms and created a resource folder:
`resources/Snippet384/Example Images`.

For detailed information and to participate in discussions about changes to the algorithm, please refer to the [accompanying discussion thread](https://github.com/eclipse-platform/eclipse.platform.swt/discussions/1741#discussion-7850360).

HeikoKlare commented on examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet384.java:
```diff
@@ -0,0 +1,418 @@
+import org.eclipse.swt.layout.*;
+import org.eclipse.swt.widgets.*;
+
+public class Snippet384 {
```
Please add a comment to the class used as the description of the snippet shown in the snippet explorer (see other snippets for a template).

HeikoKlare reviewed (commented):
@killingerm I think it would be nice to have this snippet in the repository, even though we already have multiple algorithms implemented productively now, as it allows for easier testing of alternatives and reconfigurations.

To do so, would you please squash your commits into one and sign the Eclipse Contributor Agreement (ECA)?

In addition, we need to link the new snippet in the `Snippets.md` file at the root of the `org.eclipse.swt.snippets` bundle. I would propose to add this snippet to the `Image` section of that markdown file.

akurtakov commented:
@killingerm You have to sign Eclipse contributor agreement  in order for your PR to be reviewed - https://www.eclipse.org/legal/eca/

killingerm commented:
@akurtakov I signed the ECA long ago but then changed my GitHub username to killigerm (to better reflect my name). I tried to relink it to my eclipse account but got a notification to send a mail to correct it, so far I sent two of them. I've also tried the "Eclipse ECA Validation" plugin. Is there anything else I can try?

akurtakov commented:
I'm deferring to @eclipsewebmaster as I can't help with this one.

killingerm commented:
Thank you @akurtakov, @HeikoKlare and @eclipsewebmaster, it should work now.

HeikoKlare commented:
Thank you, @killingerm! The ECA validation is fine now. Can you please address the review comments so that we can merge afterwards?

killingerm commented:
Hi again, 
@HeikoKlare I have now applied the requested changes:
- Linked the **Disabled Icon Viewer Snippet** (Snippet 348) in `Snippets.md` under the image section
- Added Javadoc describing the Snippets functionality
- Squashed everything into a single commit

I believe it has worked, though I'm not entirely sure whether the `.prefs` and `.project` files are supposed to be in the commit.

Also, since a new icon-disablement algorithm was selected I made a few adjustments:
- Added the new disabling algorithm directly underneith the original images
- Changed the order of algorithms with priority **current -> old -> experimental**
- Updated sliders default values to the ones agreed on in the discussion thread
- Replaced the preview image:

![Snippet384](https://github.com/user-attachments/assets/adc9b8a0-a52e-48cc-aec0-1c156a5e89a8)

Please have a look at it and let me know if everything is in order this way or further changes are needed!

Lastly I'm really glad the snippet was helpful and is added into the repository, thank you!

HeikoKlare reviewed (approved):
Thank you for the update! I did some minor changes. With them the PR is ready for me:
- Removed faulty additions of `.project` and preferences files
- Adapted snippet headline to be shown in SnippetExplorer
- Replaced the squashed commit message with a proper one.
